.github/workflows/CI.yaml

```diff
@@ -60,8 +72,16 @@ jobs:
         conda list
 
 
-    - name: Run tests
+    - name: Install numpy
```

Comment:
The way this workflow is laid out, installing numpy here isn't really going to work (or if anything it'll end up problematic).

You have to consider these runner workflows to be serial in time. So here you're doing the following:

1. Checking out the repo
2. Getting some info about the system (`df -h`, etc...)
3. Creating an conda environment backed on your test_env.yaml (which will auto-pull numpy, etc.. because you've either defined them or they get pulled by MDA)
4. Installed your package
5. Installed a new version of numpy
6. Ran the tests

If you want to test pinning to an older version of numpy, at least for now, my suggestion is just to create a separate environment yaml file that pins to the lowest package versions you want to support.

Then add a `env-file` entry to your runner matrix where for the normal tests it points to your normal `test_env.yaml`, and for the minimum version it points to say `test_min_env.yaml`. Then use that variable to feed the `environment-file` entry on the `conda-incubator` step.